components/pedigree/resources/src/main/resources/pedigree/filesaver/FileSaver.js

```diff
@@ -238,12 +238,13 @@ define([
   // with an attribute `content` that corresponds to the window
 
   var saveTextAs = saveTextAs
-  || (function (textContent, fileName, charset) {
+  || (function (textContent, fileName, charset, type) {
       fileName = fileName || 'pedigree.txt';
       charset = charset || 'utf-8';
       textContent = (textContent || '').replace(/\r?\n/g, "\r\n");
+      type = (type || "text/plain;charset=") + charset;
```

Comment:
I think we should just have two method, saveTextFile and savePNGFile, with hardcoded charsets inside, and no checks, since that is the functionality we need, and developers reading h code using those methods will know exactly whats happening, without caring for hairy charset details